wrangler-docs/functions/logical.md

```diff
@@ -0,0 +1,161 @@
+# Logical Bitwise
+
+The logical functions perform bit operations. All the functions described below are in the
+`logical` namespace. To use the functions defined in this namespace prefix function invocation with `logical:<func>`.
+
+## BitAnd
+Returns the bitwise `AND` of the two arguments. The inputs must be non-null.
+
+#### Namespace
+`logical`
+
+#### Input
+number1(long, int), number2(long, int)
+
+#### Output
+long
+
+#### Example
+If `a` contains the number `352` and `b` contains the number `400`,
+then the following two functions are equivalent, and return the value `256`:
+```
+set-column val logical:BitAnd(352, 400)
+set-column val logical:BitAnd(a,b)
+```
+
+## BitOr
+Returns the bitwise `OR` of the two arguments. The inputs must be non-null
+
+#### Namespace
+`logical`
+
+#### Input
+number1(long, int), number2(long, int)
+
+#### Output
+long
+
+#### Example
+If `a` contains the number `352` and `b` contains the number `400`,
+then the following two functions are equivalent, and return the value `496`.
+
+```
+set-column val logical:BitOr(352,400)
+set-column val logical:BitOr(a,b)
+```
+
+## BitXor
+Returns the bitwise `XOR` of the two arguments. The inputs must be non-null.
+
+#### Namespace
+`logical`
+
+#### Input
+number1(long, int), number2(long, int)
+
+#### Output
+long
+
+#### Example
+
+If `a` contains the number `352` and `b` contains the number `400`,
+then the following two functions are equivalent, and return the value `240`:
+
+```
+set-column val logical:BitXOr(352,400)
+set-column val logical:BitXOr(a,b)
+```
+
+## BitCompress
+Returns the `integer` made from the string argument,
+which contains a binary representation of `"1"s` and `"0"s`.
+
+#### Namespace
+`logical`
+
+#### Input
+String
+
+#### Output
+number(long)
+
+#### Example
+If `mynumber` contains the string `"101100000"`,
+then the following two functions are equivalent, and return the number `352`.
+```
+set-column val logical:BitCompress("101100000")
+set-column val logical:BitCompress(mynumber)
+```
+
+## BitExpand
+Returns a `string` containing the binary representation in `"1"s` and `"0"s`
+of the given `long`.
+
+#### Namespace
+`logical`
+
+#### Input
+number(long)
+
+#### Output
+String
+
+#### Example
+If `mynumber` contains the number `352`, then the following two functions
+are equivalent, and return the string `"101100000"`
+```
+set-column val logical:BitExpand(352)
+set-column val logical:BitExpand(mynumber)
+```
+
+## Not
+Returns the Not of the logical value of an expression. If the value of expression is `true`, the Not function
+returns a value of `false (0)`. If the value of expression is `false`, the `NOT` function returns a value of `true (1)`.
+A numeric expression that evaluates to `0` is a logical value of `false`. A numeric expression that evaluates to
+anything else, other than the `null` value, is a logical `true`. An empty string is logically `false`. All other string
```

Comment:
The numeric Not functions don't allow null. what happens if a null is passed in? Does it automatically get sent to the Not(string)?

Reply:
it's confusing to say something like 'An empty string is logically false' because you have to invert it to understand that the function actually returns true if you pass in an empty string. It would be more clear to just say what the function returns given various input.

Reply:
null will get sent to  Not(string)

Reply:
Fixed the description